lib/Runtime/Library/JavascriptArray.cpp

```diff
@@ -11935,7 +11934,7 @@ using namespace Js;
 
         Assert(IsInlineSegment(src, instance) == IsInlineSegment(dst, static_cast<T*>(this)));
 
-        CopyArray(dst->elements, dst->size, src->elements, src->size);
+        CopyArray(dst->elements, dst->size, src->elements, dst->size);
```

Comment:
Unless I'm misunderstanding, I think this should still be src->size. The point of this is that memcpy will fail (and not read OOB) if dst->size > src->size.

Reply:
dst->size will be <= src->size here. if you see above
```dst->size = min(src->size, inlineChunkSize);```
I changed this as I was getting assert in this function. What I can do is to put ```Assertorfailfast(dst->size <=src->size)```.


---
In reply to: [274567842](https://github.com/Microsoft/ChakraCore/pull/6084#discussion_r274567842) [](ancestors = 274567842)

Reply:
ok, that works for me

Reply:
resolved this.